### ISSUE TITLE
Crash on sharing a file

### ISSUE THREAD

**Issue by mikini** [nextcloud/android#8083]
### Steps to reproduce
1. Pick a random file in Files
2. Click share icon
3. Search for a local user
4. Choose local user when popping up
5. Click "Share"

### Expected behaviour
- File should be shared with chosen user

### Actual behaviour
- A wait spinner appears, then application  crashes

### Can you reproduce this problem on https://try.nextcloud.com?
- Please create a test demo account and see if this still happens there.

I'm either logged into a Trial Account in which I can't establish a password to use in the application or I'm asked to contact you for a quote. How am I supposed to do this?

### Environment data
Android version:
9

Device model: 
Samsung Galaxy S8

Stock or customized system:
Stock

Nextcloud app version:
30150090 (latest on F-Droid)

Nextcloud server version:
20-something (latest from Syncloud)

Reverse proxy:
No

#### Web server error log
Lost in SDcard formatting

#### Nextcloud log (data/nextcloud.log)
Lost SDCard formatting
 
#### Android application  stack trace

```
************ CAUSE OF ERROR ************

java.lang.RuntimeException: Accessing result data after operation failed!
 at 
com.owncloud.android.lib.common.operations.RemoteOperationResult.getData(RemoteOperationResult.java:486)
 at 
com.owncloud.android.utils.ErrorMessageAdapter.getMessageForCreateShareOperations(ErrorMessageAdapter.java:254)
 at 
com.owncloud.android.utils.ErrorMessageAdapter.getMessageForResultAndOperation(ErrorMessageAdapter.java:143)
 at 
com.owncloud.android.utils.ErrorMessageAdapter.getErrorCauseMessage(ErrorMessageAdapter.java:79)
 at 
com.owncloud.android.ui.activity.FileActivity.onRemoteOperationFinish(FileActivity.java:380)
 at 
com.owncloud.android.ui.activity.FileDisplayActivity.onRemoteOperationFinish(FileDisplayActivity.java:1704)
 at 
com.owncloud.android.services.OperationsService.lambda$dispatchResultToOperationListeners$0(OperationsService.java:734)
 at 
com.owncloud.android.services.-$$Lambda$OperationsService$KsB1LbjHjlEBSp0h7XO24czs-OQ.run(Unknown 
Source:6)
 at android.os.Handler.handleCallback(Handler.java:873)
 at android.os.Handler.dispatchMessage(Handler.java:99)
 at android.os.Looper.loop(Looper.java:214)
 at android.app.ActivityThread.main(ActivityThread.java:7050)
 at java.lang.reflect.Method.invoke(Native Method)
 at 
com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:965)

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30150090
Build flavor: generic

************ DEVICE INFORMATION ************
Brand: samsung
Device: dreamlte
Model: SM-G950F
Id: PPR1.180610.011
Product: dreamltexx

************ FIRMWARE ************
SDK: 28
Release: 9
Incremental: G950FXXSBDTJ1
```

**Comment by tobiasKaminsky** [comment]
Please try 3.15.1 RC1 via Play store beta and re-open if it still happens.

**Comment by mikini** [comment]
I can confirm that this is fixed in 3.15.1 RC1.

Had to wait for the version becoming available in FD, was not able to fetch the beta from Play on top of the installed version from FD even after registering for beta. Play seemed to think I already had RC1 of some reason.
